Observer/Adminhtml/Config.php

```diff
@@ -0,0 +1,114 @@
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * @author Nosto Solutions Ltd <contact@nosto.com>
+ * @copyright 2019 Nosto Solutions Ltd
+ * @license http://opensource.org/licenses/BSD-3-Clause BSD 3-Clause
+ *
+ */
+
+namespace Nosto\Tagging\Observer\Adminhtml;
+
+use Magento\Framework\Event\Observer;
+use Magento\Framework\Event\ObserverInterface;
+use Magento\Framework\Module\Manager as ModuleManager;
+use Nosto\Tagging\Helper\Data as NostoHelperData;
+use Nosto\Tagging\Helper\Scope as NostoHelperScope;
+use Nosto\Tagging\Logger\Logger as NostoLogger;
+use Nosto\Tagging\Model\ResourceModel\Product\Index\Collection as IndexCollection;
+use Nosto\Tagging\Helper\Account as NostoAccountHelper;
+
+/**
+ * Observer to mark all indexed products as dirty if settings have changed
+ *
+ * @package Nosto\Tagging\Observer\Adminhtml
+ */
+class Config implements ObserverInterface
+{
+    private $logger;
+    private $moduleManager;
+    private $nostoHelperScope;
+    private $nostoAccountHelper;
+    private $indexCollection;
+
+    /**
+     * Config Constructor.
+     *
+     * @param NostoLogger $logger
+     * @param ModuleManager $moduleManager
+     * @param NostoHelperScope $nostoHelperScope
+     * @param NostoAccountHelper $nostoAccountHelper
+     * @param IndexCollection $indexCollection
+     */
+    public function __construct(
+        NostoLogger $logger,
+        ModuleManager $moduleManager,
+        NostoHelperScope $nostoHelperScope,
+        NostoAccountHelper $nostoAccountHelper,
+        IndexCollection $indexCollection
+    ) {
+        $this->logger = $logger;
+        $this->moduleManager = $moduleManager;
+        $this->nostoHelperScope = $nostoHelperScope;
+        $this->nostoAccountHelper = $nostoAccountHelper;
+        $this->indexCollection = $indexCollection;
+    }
+
+    /**
+     * Observer method to mark all indexed products as dirty on the index table
+     *
+     * @param Observer $observer the dispatched event
+     */
+    public function execute(Observer $observer) // @codingStandardsIgnoreLine
+    {
+        if (!$this->moduleManager->isEnabled(NostoHelperData::MODULE_NAME)) {
+            return;
+        }
+        $changedConfig = $observer->getData('changed_paths');
+        if (!empty($changedConfig) && !in_array(NostoHelperData::XML_PATH_INDEXER_MEMORY, $changedConfig, false)) {
+            $store = $this->nostoHelperScope->getStore($observer->getData('store'));
```

Comment:
What if the settings are changed in global scope? Does the observer multiple times or?